contrib/config_option_database/utils/Yacc2Graph.py

```diff
@@ -0,0 +1,131 @@
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+import xml.etree.ElementTree as xml_parser
+from os import remove
+from subprocess import DEVNULL, Popen
+from tempfile import NamedTemporaryFile
+
+import networkx
+
+
+class Rule():
+    def __init__(self, number, parent, symbols):
+        self.number = number
+        self.parent = parent
+        self.symbols = symbols
+
+
+def _run_in_shell(command):
+    proc = Popen(command, stderr=DEVNULL, stdout=DEVNULL)
+    proc.wait()
+    return proc.returncode == 0
+
+
+def _write_to_file(string):
+    file = NamedTemporaryFile(mode='w')
+    file.write(string)
+    file.flush()
+    file.seek(0)
+    return file
+
+
+def _yacc2xml(yacc):
+    with _write_to_file(yacc) as file:
+        output = NamedTemporaryFile().name
```

Comment:
Renamed it to `xml_filepath`, because it suits it better, I believe. :)
Here we generate a temp file name, and ask bison to generate the xml-file there, so it is not temporary. It will be removed later explicitly.